pkgs/by-name/mo/mouse-actions/package.nix

```
@@ -0,0 +1,49 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+, pkg-config
+, libX11
+, libXi
+, libXtst
+, libevdev
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "mouse-actions";
+  version = "0.4.4";
+
+  src = fetchFromGitHub {
+    owner = "jersou";
+    repo = "mouse-actions";
+    rev = "v${version}";
+    hash = "sha256-02E4HrKIoBV3qZPVH6Tjz9Bv/mh5C8amO1Ilmd+YO5g=";
+  };
+
+  cargoHash = "sha256-5SUVZlrXIPtlu9KBzucZDCp5t5t8Z4/Nfht2Pw5agVI=";
+
+  buildInputs = [
+    libX11
+    libXi
+    libXtst
+    libevdev
+  ];
```

I'm not familiar with darwin, but this seems reasonable based on https://github.com/jersou/mouse-actions/blob/25b341c0bd8627e3a7bf0dacc7b01e7664369028/rdev/Cargo.toml#L25-L37

I think you'll have to add the darwin dependencies through an entry in `pkgs/top-level/all-packages.nix`

@wegank do you mind if I ping you for this?

```suggestion
  buildInputs = [ ]
    ++ (lib.optionals stdenv.isLinux [
      libX11
      libXi
      libXtst
      libevdev
    ])
    ++ (lib.optionals stdenv.isDarwin [
      Cocoa
      CoreGraphics
      CoreFoundation
    ]);
```

---

Adding `darwin.apple_sdk.frameworks.Cocoa` should suffice. However, I had to

- replace `Pid::from_raw(pid)` with `Pid::from_raw(pid as i32)` in `src/single_instance.rs` for the build to succeed, and,
- make the `assert_eq!` on floats looser in `test_points_to_angles`.

I wonder if upstream has really tested darwin support.

---

the author mentions that mac __should__ work but it looks like they don't have a mac machine to test so I wouldn't be so sure about it.

> ## Platform compatibility
> I only tested it on Linux + X11 and Linux + Wayland, but it should work on Mac, Windows.

---

Hi, sorry I didn't see this until now. I would definitely be willing to add these changes.

But, I am unsure what the protocol is- am I responsible for testing/patching this package on mac, just for nixpkgs, when the upstream seems uninsterested/unable to test for themselves?

Moreover, it seems that [macOS does not have udev rules](https://forums.developer.apple.com/forums/thread/79167), which according to upstream is a [necessary](https://github.com/jersou/mouse-actions?tab=readme-ov-file#requirement-) feature to use the program. 

Open to advice!

---

If upstream isn't supporting darwin then I'd say mark it as only supporting linux for now. To say it supports "all" platforms is a bit misleading

---

Alright, it seems that everyone in this thread agrees this package is effectively linux-only. I'll mark it as resolved.